framework/src/userobject/SolutionUserObject.C

```diff
@@ -645,6 +645,8 @@ SolutionUserObject::evalMeshFunction(const Point & p, std::string var_name, unsi
 
   // Extract the variable index for the MeshFunction(s), must use iterator b/c of const
   std::map<std::string, unsigned int>::const_iterator it = _local_variable_index.find(var_name);
+  if (it == _local_variable_index.end())
```

Comment:
This doesn't seem right. Why can't we detect this in the ctor? Maybe we just need to use the "variables" parameter to "VariableName"? I'd have to dig in bit more but this just looks way too late.